tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -782,3 +791,343 @@ def test_tag_object_without_permission(self, user_attr, expected_status):
+        response = self.client.get(self.small_taxonomy_url)
+        assert response.status_code == status.HTTP_200_OK
+
+        data = response.data
+        results = data.get("results", [])
+
+        # Count of root tags
+        root_count = self.small_taxonomy.tag_set.filter(parent=None).count()
+        assert len(results) == root_count
+
+        # Checking tag fields
+        root_tag = self.small_taxonomy.tag_set.get(id=results[0].get("id"))
+        root_children_count = root_tag.children.count()
+        assert results[0].get("value") == root_tag.value
+        assert results[0].get("taxonomy_id") == self.small_taxonomy.id
+        assert results[0].get("children_count") == root_children_count
+        assert len(results[0].get("sub_tags")) == root_children_count
+
+        # Checking pagination values
+        assert data.get("next") is None
+        assert data.get("previous") is None
+        assert data.get("count") == root_count
+        assert data.get("num_pages") == 1
+        assert data.get("current_page") == 1
+        assert data.get("left_index") == 1
+        assert data.get("right_index") == root_count
+
+    def test_small_search(self):
+        search_term = 'eU'
+        url = f"{self.small_taxonomy_url}?search_term={search_term}"
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+
+        data = response.data
+        results = data.get("results", [])
+
+        assert len(results) == 3
+
+        # Checking pagination values
+        assert data.get("next") is None
+        assert data.get("previous") is None
+        assert data.get("count") == 3
+        assert data.get("num_pages") == 1
+        assert data.get("current_page") == 1
+        assert data.get("left_index") == 1
+        assert data.get("right_index") == 3
+
+    def test_large_taxonomy(self):
+        self._build_large_taxonomy()
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.get(self.large_taxonomy_url)
+        assert response.status_code == status.HTTP_200_OK
+
+        data = response.data
+        results = data.get("results", [])
+
+        # Count of paginated root tags
+        assert len(results) == self.page_size
+
+        # Checking tag fields
+        root_tag = self.large_taxonomy.tag_set.get(id=results[0].get("id"))
+        assert results[0].get("value") == root_tag.value
+        assert results[0].get("taxonomy_id") == self.large_taxonomy.id
+        assert results[0].get("parent_id") == root_tag.parent_id
+        assert results[0].get("children_count") == root_tag.children.count()
+        assert (
+            results[0].get("sub_tags_link")
+            ==
+            (
+                "http://testserver/tagging/"
+                f"rest_api/v1/taxonomies/{self.large_taxonomy.id}"
+                f"/tags/?parent_tag_id={root_tag.id}"
+            )
+        )
```

Comment:
Nit: This formatting is a bit strange with `==` on its own line. Why not

```python
assert results[0].get("sub_tags_link") == (
    ...
)
```
like you have below? I find that much more readable.

Reply:
Updated here: https://github.com/openedx/openedx-learning/pull/78/commits/9a15562c8509eee795dff0a939a9310b8a33af3f